Cesium.CodeGen/Ir/Expressions/SubscriptingExpression.cs

```diff
@@ -31,12 +31,14 @@ public IExpression Lower()
     public void EmitTo(IDeclarationScope scope) => Resolve(scope).EmitGetValue(scope);
 
     public IType GetExpressionType(IDeclarationScope scope) => Resolve(scope).GetValueType();
-
+    
     public IValue Resolve(IDeclarationScope scope)
     {
-        if (_expression is not IdentifierExpression identifier)
-            throw new WipException(230, "Subscription supported only for IdentifierConstantExpression");
-
-        return new LValueArrayElement(identifier.Resolve(scope), _index);
+        return _expression switch
+        {
+            IdentifierExpression identifier => new LValueArrayElement(identifier.Resolve(scope), _index),
+            IValueExpression valueExpression => valueExpression.Resolve(scope),

```

Comment:
Now wait a minute, something's missing here. It looks like the the `_index` is ignored in this case? It shouldn't be.

And please add a unit test (to, say, `CodeGenMethodTests` class).